### PR TITLE
Backend plugin

### PR DESCRIPTION
**Pull request by piotr-signalfx**

### CODE REVIEW

File: pkg/job_handler.go [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+		return float64(i)
+	default:
+		return 0.0
+	}
+}
+
+func (t *SignalFxJobHandler) stop() {
+	if t.running {
+		t.logger.Debug("Stopping job", "program", t.program)
+		t.computation.Stop()
```
Comment (keitwb) [inline pkg/job_handler.go]: Probably best to guard for `t.compuation == nil` to avoid panics just in case the handler never got fully initialized.

Comment (keitwb) [reply pkg/job_handler.go]: Actually, this whole method can probably just be:

```
if t.computation != nil {
    t.computation.Stop()
    t.compuatation = nil
}
```

It is idempotent to call Stop multiple times.  Obvious a mutex will be necessary if stop needs to be reentrant.

File: pkg/job_handler.go [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+	series := make([]*datasource.TimeSeries, 0)
+	for id, points := range t.Points {
+		s := &datasource.TimeSeries{Name: t.getTimeSeriesName(id), Points: points}
+		series = append(series, s)
+	}
+	return series
+}
+
+// TODO: Use metadata to map tsid onto some meaningful name
+func (t *SignalFxJobHandler) getTimeSeriesName(tsid int64) string {
```
Comment (keitwb) [inline pkg/job_handler.go]: This is possible now.  You would want the `OriginatingMetric` field on the MetadataMessage struct for that TSID.

File: pkg/job_handler.go (hunk 1 above)
Comment (piotr-signalfx) [reply pkg/job_handler.go]: Setting computation to nil would prevent us from re-using job handlers for fixed time ranges as we still need it to get the resolution for instance in trimDatapoints().

Comment (piotr-signalfx) [reply pkg/job_handler.go]: Do we need `t.compuation == nil`  despite `t.running` being set to true after the value is assigned to `t.computation` ?

File: pkg/job_handler.go (hunk 2 above)
Comment (keitwb) [reply pkg/job_handler.go]: I went ahead and did this here.

File: pkg/job_handler.go (hunk 1 above)
Comment (keitwb) [reply pkg/job_handler.go]: I just removed the `t.running` flag since it didn't seem to have any purpose since `t.computation.Stop` is idempotent.

**Pull request merged**